pages/signup.js

```diff
@@ -19,6 +19,12 @@ import { auth } from '../utils/api/firebase.config'
 import { useAuthContext } from '../utils/context/AuthProvider';
 
 export default function Signup() {
+  const [googleOpen, setGoogleOpen] = useState(false);
+  const [facebookOpen, setFacebookOpen] = useState(false);
+  const handleGoogleOpen = () => setGoogleOpen(true);
+  const handleFacebookOpen = () => setFacebookOpen(true);
+  const handleGoogleClose = () => setGoogleOpen(false);
+  const handleFacebookClose = () => setFacebookOpen(false);
```

Comment:
Could be clearer with naming. googleOpen and facebookOpen can be more descriptive. Could use "googleSignUpModal"